core/procedures.js

```diff
@@ -358,9 +424,58 @@ Blockly.Procedures.editProcedureCallback_ = function(block) {
       return;
     }
     block = innerBlock;
+  } else if (!block.type == Blockly.PROCEDURES_PROTOTYPE_BLOCK_TYPE){
+    // This is a call block, find the prototype corresponding to the procCode
+    block = Blockly.Procedures.getPrototypeBlock(
+      block.getProcCode(), block.workspace);
   }
-  alert('TODO(#603): implement function editing (procCode was "' +
-      block.procCode_ + '")');
+
+  // Block now refers to the procedure prototype block, it is safe to proceed.
+  Blockly.Procedures.externalProcedureDefCallback_(
+    block.mutationToDom(),
+    Blockly.Procedures.editProcedureCallbackFactory_(block)
+  );
+};
+
+/**
+ * Callback factory for editing an existing custom procedure.
+ * @param {!Blockly.Block} block The procedure prototype block being edited.
+ * @return {function(?Element)} Callback for editing the custom procedure.
+ * @private
+ */
+Blockly.Procedures.editProcedureCallbackFactory_ = function(block) {
+  return function(mutation) {
+    if (mutation) {
```

Comment:
This belongs in Blockly.Procedures.mutateCallers: https://github.com/LLK/scratch-blocks/blob/develop/core/procedures.js#L311